sql/database.go

```diff
@@ -51,3 +53,39 @@ func (p *planner) getDatabaseDesc(name string) (*DatabaseDescriptor, error) {
 	}
 	return &desc, nil
 }
+
+// getCachedDatabaseDesc looks up the database descriptor given its name in the
+// descriptor cache.
+func (p *planner) getCachedDatabaseDesc(name string) (*DatabaseDescriptor, error) {
+	if name == SystemDB.Name {
+		return &SystemDB, nil
+	}
+
+	nameKey := databaseKey{name}
+	var nameVal *roachpb.Value
+	if p.systemConfig != nil {
+		nameVal = p.systemConfig.GetValue(nameKey.Key())
+	}
+	if nameVal == nil {
+		return nil, fmt.Errorf("database %q does not exist in system cache", nameKey.Name())
+	}
+
+	id, err := nameVal.GetInt()
+	if err != nil {
+		return nil, err
+	}
+
+	descKey := MakeDescMetadataKey(ID(id))
+	descVal := p.systemConfig.GetValue(descKey)
+	if descVal == nil {
+		return nil, fmt.Errorf("database %q has name entry, but no descriptor in system cache",
```

Comment:
This error shouldn't happen, though I haven't been able to prove to myself it can't happen so I'd like to leave this as `fmt.Errorf`. The value of `util.Errorf` here is minor has the error message will lead right to this line of code. The earlier `fmt.Errorf` can definitely happen if the user references a database that doesn't exist.